src/Featurizers/Structs.h

```diff
@@ -120,14 +123,28 @@ SparseVectorEncoding<T>::ValueEncoding::ValueEncoding(value_type value, std::uin
     Index(std::move(index)) {
 }
 
+template <typename T>
+SparseVectorEncoding<T>::ValueEncoding::ValueEncoding(ValueEncoding && other) :
+    Value(std::move(const_cast<value_type &>(other.Value))),
+    Index(std::move(const_cast<std::uint64_t &>(other.Index))) {
+}
+
+template <typename T>
+typename SparseVectorEncoding<T>::ValueEncoding & SparseVectorEncoding<T>::ValueEncoding::operator =(ValueEncoding && other) {
+    const_cast<value_type &>(this->Value) = std::move(const_cast<value_type &>(other.Value));
+    const_cast<std::uint64_t &>(this->Index) = std::move(const_cast<std::uint64_t &>(other.Index));
+
+    return *this;
+}
+
 #if (defined __clang__)
 #   pragma clang diagnostic push
 #   pragma clang diagnostic ignored "-Wfloat-equal"
 #endif
 
 template <typename T>
 bool SparseVectorEncoding<T>::ValueEncoding::operator==(ValueEncoding const &other) const {
-    return Value == other.Value
+    return static_cast<T>(abs(Value - other.Value)) < static_cast<T>(EPS)

```

Comment:
If you make this change here, we need to make similar changes to all equality operators. Can the Catch2 Approx functionality give you what you need for now?

See SharedLibrary_NumericalizerFeaturizer_IntegrationTest.cpp for an example.

Reply:
yeah I can use that to customize my test function. I'll close this PR